There is a crash in Qt when a certain Animator combination is removed from a window,
the window is destroyed and then the item is inserted into a new window.

https://qt-project.atlassian.net/browse/QTBUG-150367

BUG: 487699


(cherry picked from commit 20f6761bb4dcb52ecf4929cdb05171931c4a8244)

677b1765 systemtray: Port away from Animator

Co-authored-by: David Edmundson <kde@davidedmundson.co.uk>
---
 applets/systemtray/qml/AbstractItem.qml   | 2 +-
 applets/systemtray/qml/PulseAnimation.qml | 8 +++++---
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/applets/systemtray/qml/AbstractItem.qml b/applets/systemtray/qml/AbstractItem.qml
index b3a80da95e..16f4039d1d 100644
--- a/applets/systemtray/qml/AbstractItem.qml
+++ b/applets/systemtray/qml/AbstractItem.qml
@@ -116,7 +116,7 @@ PlasmaCore.ToolTipArea {
             Accessible.onPressAction: abstractItem.activated(Plasmoid.popupPosition(iconContainer, iconContainer.width/2, iconContainer.height/2));
 
             Behavior on scale {
-                ScaleAnimator {
+                NumberAnimation {
                     duration: Kirigami.Units.longDuration
                     easing.type: (abstractItem.effectivePressed || mouseArea.containsPress) ? Easing.OutCubic : Easing.InCubic
                 }
diff --git a/applets/systemtray/qml/PulseAnimation.qml b/applets/systemtray/qml/PulseAnimation.qml
index c81f8de9bb..d1bbc09df2 100644
--- a/applets/systemtray/qml/PulseAnimation.qml
+++ b/applets/systemtray/qml/PulseAnimation.qml
@@ -20,17 +20,19 @@ SequentialAnimation {
 
     loops: Animation.Infinite
     alwaysRunToEnd: true
-
-    ScaleAnimator {
+    // Avoiding an animator due to https://qt-project.atlassian.net/browse/QTBUG-150367
+    NumberAnimation {
         target: pulseAnimation.targetItem
+        property: "scale"
         from: 1
         to: 1.2
         duration: pulseAnimation.duration * 0.15
         easing.type: Easing.InQuad
     }
 
-    ScaleAnimator {
+    NumberAnimation {
         target: pulseAnimation.targetItem
+        property: "scale"
         from: 1.2
         to: 1
         duration: pulseAnimation.duration * 0.15
-- 
GitLab

